src/core/MOM_stoch_eos.F90

```diff
@@ -0,0 +1,212 @@
+!> Provides the ocean stochastic equation of state
+module MOM_stoch_eos
+! This file is part of MOM6. See LICENSE.md for the license.
+use MOM_grid,            only : ocean_grid_type
+use MOM_hor_index,       only : hor_index_type
+use MOM_file_parser,     only : get_param, param_file_type
+use MOM_random,          only : PRNG,random_2d_constructor,random_2d_norm
+use MOM_time_manager,    only : time_type
+use MOM_io,              only : vardesc, var_desc
+use MOM_restart,         only : MOM_restart_CS,is_new_run
+use MOM_diag_mediator,   only : register_diag_field,post_data,diag_ctrl,safe_alloc_ptr
+use MOM_variables,       only : thermo_var_ptrs
+use MOM_verticalGrid,    only : verticalGrid_type
+use MOM_restart,         only : register_restart_field
+use MOM_isopycnal_slopes,only : vert_fill_TS
+!use random_numbers_mod, only : getRandomNumbers,initializeRandomNumberStream,randomNumberStream
+
+implicit none
+#include <MOM_memory.h>
+
+public MOM_stoch_eos_init
+public MOM_stoch_eos_run
+public MOM_calc_varT
+
+real,private ALLOCABLE_, dimension(NIMEM_,NJMEM_) :: l2_inv
```

Comment:
The MOM6 style guide strongly discourages the use of module data (see https://github.com/mom-ocean/MOM6/wiki/Code-style-guide#global--module-data). Module data precludes on-line ensembles, and it promotes horrible code practices like not passing in the actual data as arguments, making it very hard to figure out where information is coming from.  There are very limited exceptions where this is permitted (e.g, clock IDs that measure the time spent on a line of code, which should be summed across all ensemble members), but this case does not appear to match any justifiable exceptions.

  Please put these variables into an opaque control structure for this module, and if they need to be accessed outside of this module, we should provide an inquire routine to obtain them.